api/base/authentication/drf.py

```diff
@@ -112,9 +117,11 @@ def authenticate(self, request):
         if cas_auth_response.authenticated is False:
             raise exceptions.NotAuthenticated(_('CAS server failed to authenticate this token'))
 
-        user_id = cas_auth_response.user
-        user = User.load(user_id)
-        if user is None:
+        # TODO: @cslzchen use GUID when switching from OSF-Postgres-CAS to OSF-API-CAS
```

Comment:
I choose a simple fix here sine `username` is an unique property.

However, this will be further improved when CAS talks to Database through OSF API instead of direct access.